packages/dds/merge-tree/src/client.ts

```diff
@@ -346,6 +350,49 @@ export class Client {
         return this.mergeTree.getMarkerFromId(id);
     }
 
+    /**
+     * Revert an op
+     */
+    public rollback?(op: any, localOpMetadata: unknown) {
+        if (op.type !== MergeTreeDeltaType.INSERT) {
+            throw new Error("Unsupported op type for rollback");
+        }
+
+        const lastList = this.mergeTree.pendingSegments?.prev;
+        if (!lastList) {
+            throw new Error("No pending segments");
+        }
+        ListRemoveEntry(lastList);
+        const pendingSegmentGroup = lastList.data;
+        if (pendingSegmentGroup === undefined || pendingSegmentGroup !== localOpMetadata) {
+            throw new Error("Rollback op doesn't match last edit");
+        }
+        let length = 0;
+        for (const segment of pendingSegmentGroup.segments) {
+            const segmentSegmentGroup = segment.segmentGroups.dequeue();
+            assert(segmentSegmentGroup === pendingSegmentGroup, "Unexpected segmentGroup in segment");
+            if (TextSegment.is(segment)) {
+                length += segment.text.length;
+            } else if (Marker.is(segment)) {
+                length += 1;
+            } else {
+                throw new Error("Unsupported segment type for rollback");
+            }
+        }
+
+        const start = op.pos1 as number;
+        const segWindow = this.getCollabWindow();
+        const removeOp = createRemoveRangeOp(start, start + length);
+        this.mergeTree.markRangeRemoved(
+            start,
+            start + length,
+            UniversalSequenceNumber,
+            segWindow.clientId,
+            TreeMaintenanceSequenceNumber,
```

Comment:
i don't think this constant is actually safe to use in this case, as we never expect it on a segment, it only goes into the partial length structure. using it here will put it as the removedSeq. i think UniversalSequenceNumber would be a better fit.